### PR TITLE
Convert a bunch of subprojects build scripts to Kotlin

### PR DESCRIPTION
**Pull request by eskatos**
- `:*-internal-*-testing`
- the 6 smaller remaining

Part of #4649

### CODE REVIEW

**Comment by eskatos** [comment]
Feedback build https://builds.gradle.org/viewQueued.html?itemId=11520427

**Comment by eskatos** [comment]
New feedback build https://builds.gradle.org/viewQueued.html?itemId=11789389

File: buildSrc/subprojects/kotlin-dsl/src/main/kotlin/accessors/accessors.kt [hunk 1]
```diff
@@ -56,3 +59,7 @@ fun Project.idea(configure: IdeaModel.() -> Unit): Unit =
 
 fun Project.eclipse(configure: EclipseModel.() -> Unit): Unit =
     extensions.configure("eclipse", configure)
+
+
+val RepositoryHandler.javaScript
+    get() = (this as ExtensionAware).extensions.getByType(JavaScriptRepositoriesExtension::class.java)
```
Comment (bamboo) [inline buildSrc/subprojects/kotlin-dsl/src/main/kotlin/accessors/accessors.kt]: :x: Reuse `the<T>()` here?

```kotlin

val RepositoryHandler.javaScript
    get() = (this as ExtensionAware).the<JavaScriptRepositoriesExtension>()
```

**Comment by eskatos** [comment]
@bamboo ready for another round

**Pull request merged**